Update local main branch during `git update`

f1sherman opened the pull request:
This ensures that the local main branch is updated during `git update`. Without this it's necessary for the end user to run `git update` separately on the main branch, and it can be confusing if they e.g. want to diff their newly-updated feature branch against the main branch and don't realize it's stale. This happens to me fairly often and I couldn't think of a reason someone _wouldn't_ want their main branch updated so this seemed like a sane change.

f1sherman committed:
Update local main branch during `git update`

This ensures that the local main branch is updated during `git update`.
Without this it's necessary for the end user to run `git update`
separately on the main branch, and it can be confusing if they e.g. want
to diff their newly-updated feature branch against the main branch and
don't realize it's stale.

wireframe commented:
> I couldn't think of a reason someone _wouldn't_ want their main branch updated so this seemed like a sane change.

+1. This is a common paper cut for me as well.  I love this change.

One question...
instead of updating the base branch at the end of the operation, this could be inverted to... (pseudocode)
- update base branch
- update feature branch from remote
- merge local base branch into feature branch

Thoughts?

f1sherman committed:
Update main branch before feature branch

Per @wireframe's recommendation, update the main branch from remote
before updating the feature branch, and update the feature branch from
local main.

f1sherman commented:
@wireframe love it! Done in https://github.com/wireframe/gitx/pull/64/commits/be91d7b9ed54a43877e653325ef4c6a34b5b802f.
